components/hab/src/command/studio/enter.rs

```diff
@@ -132,10 +133,29 @@ mod inner {
         return false;
     }
 
-    fn rerun_with_sudo_if_needed(ui: &mut UI) -> Result<()> {
-        // If I have root permissions, early return, we are done.
+    fn has_docker_group() -> bool {
+        let groupname = "docker".to_string();
+        let current_user = group::get_current_username().unwrap();
+        let docker_members = group::get_members_by_groupname(&groupname);
+        let mut _ret = false;
+        for user in docker_members {
+            if user == current_user {
+                _ret = true;
+                break;
+            }
+        }
+        _ret
```

Comment:
You can replace everything from lines 140-147 with something like:

```rust
docker_members.contains(&current_user)
```

That's with the code as it is now.  If you take my suggestions on https://github.com/habitat-sh/core/pull/16 and make `get_members_by_groupname` return an `Option`, then this would change to something like:

```rust
docker_members.map_or(false, |d| d.contains(&current_user))
```